runtime/src/identity.rs

```diff
@@ -568,6 +492,212 @@ decl_module! {
+
+            // Remove any Pre-Authentication & link
+            signers_to_remove.iter().for_each( |signer| {
+                Self::remove_pre_join_identity( signer, did);
+                if let Signer::Key(ref key) = signer {
+                    Self::unlink_key_to_did(key, did);
+                }
+            });
+
+            // Update signing keys at Identity.
+            <DidRecords>::mutate(did, |record| {
+                (*record).remove_signing_items( &signers_to_remove);
+            });
+
+            Self::deposit_event(RawEvent::RevokedSigningItems(did, signers_to_remove));
+            Ok(())
+        }
+
+        /// It adds signing keys to target identity `id`.
+        /// Keys are directly added to identity because each of them has an authorization.
+        ///
+        /// Arguments:
+        ///     - `origin` Master key of `id` identity.
+        ///     - `id` Identity where new signing keys will be added.
+        ///     - `additional_keys` New signing items (and their authorization data) to add to target
+        ///     identity.
+        ///
+        /// Failure
+        ///     - It can only called by master key owner.
+        ///     - Keys should be able to linked to any identity.
+        pub fn add_signing_items_with_authorization( origin,
+                id: IdentityId,
+                additional_keys: Vec<SigningItemWithAuth>) -> Result {
+            let sender = ensure_signed(origin)?;
+            let sender_key = Key::try_from(sender.encode())?;
+            let _grants_checked = Self::grant_check_only_master_key(&sender_key, id)?;
+
+            let authorization = TargetIdAuthorization {
+                target_id: id,
+                nonce: <system::Module<T>>::account_nonce(sender)
+            }.encode();
+
+            // 1. Verify signatures.
+            for si_with_auth in additional_keys.iter() {
+                let si = &si_with_auth.signing_item;
+
+                // Get account_id from signer
+                let account_id_found = match si.signer {
+                    Signer::Key(ref key) =>  Public::try_from(key.as_slice()).ok(),
+                    Signer::Identity(ref id) if <DidRecords>::exists(id) => {
+                        let master_key = <DidRecords>::get(id).master_key;
+                        Public::try_from( master_key.as_slice()).ok()
+                    },
+                    _ => None
+                };
+
+                if let Some(account_id) = account_id_found {
+                    if let Signer::Key(ref key) = si.signer {
+                        ensure!( Self::can_key_be_linked_to_did( key, si.signer_type),
+                        "One signing key can only belong to one identity");
+                    }
+
+                    let signature = AnySignature::from( Signature::from_h512(si_with_auth.auth_signature));
+                    ensure!( signature.verify( authorization.as_slice(), &account_id),
```

Comment:
Note: Multisigs (in their current implementation) are unable to sign any data and therefore, they can not be added with this process as signing items.